### PR TITLE
feat: capture reason for Wasm panics in CLI

### PR DESCRIPTION
**Pull request by jkomyno**
TypeScript implementation of https://github.com/prisma/prisma-private/issues/212.
Companion of the Rust PR https://github.com/prisma/prisma-engines/pull/3710 (see its description for a thorough example).

This PR fixes the Wasm limitation of not transmitting the reason and code location of a Panic by explicitly sending this panic info to a globally available JS object, `globalThis.WASM_PANIC_REGISTRY`.

### CODE REVIEW

File: packages/internals/typings/global.d.ts
Comment (jkomyno) [inline packages/internals/typings/global.d.ts:5]:
```suggestion
  // eslint-disable-next-line no-var
  var WASM_PANIC_REGISTRY: WasmPanicRegistry
```

File: packages/internals/src/WasmPanicRegistry.ts
Comment (aqrln) [inline packages/internals/src/WasmPanicRegistry.ts:2]: It may not be currently safe to use private fields, depending on whether they end up in typings: https://prisma-company.slack.com/archives/CUXLS0Z6K/p1675164034311419. Could be okay here though. Better to run ecosystem-tests before merging to make sure.

File: packages/internals/src/wasm.ts
Comment (aqrln) [inline packages/internals/src/wasm.ts:17]: I'm not very comfortable with this name as it's too generic and could collide with something with an incompatible API if end users or maintainers of another library take a similar approach in solving the same problem. How about `PRISMA_WASM_PANIC_REGISTRY`?

Since this would need to be changed on the Rust side as well, it's totally okay to do that in follow-up PRs if we want to merge this one ASAP to unbreak `main`, but it's important that we do it before the release.

Comment (jkomyno) [reply packages/internals/src/wasm.ts:17]: Agree, let's do this before the release, but after this PR.

File: packages/internals/src/WasmPanicRegistry.ts
Comment (jkomyno) [reply packages/internals/src/WasmPanicRegistry.ts]: I've simply removed the `#` prefix and replaced it with `private`.

**Review by aqrln — APPROVED** [review]

File: packages/internals/src/wasm.ts
Comment (jkomyno) [reply packages/internals/src/wasm.ts:19]: Follow-up PR for Rust: https://github.com/prisma/prisma-engines/pull/3722

**Pull request merged**
